src/main/scala/io/scalac/lab/api/endpoints4s/ApartmentsEndpointsDefinition.scala

```diff
@@ -26,41 +27,38 @@ trait ApartmentsEndpointsDefinition
   private implicit val addressSchema: JsonSchema[Address] = genericJsonSchema
   private implicit val apartmentSchema: JsonSchema[Apartment] = genericJsonSchema
 
-  val listApartments: Endpoint[(Paging, ApiKey), Either[ApiError, List[Apartment]]] =
+  val listApartments: Endpoint[(Paging, ApiKey), Either[BadRequestError, List[Apartment]]] =
     authenticatedEndpoint(
       request = get(path / "v1" / "data" / "apartments" /? pagingQueryString),
-      response = withStatusCodes(ok(jsonResponse[List[Apartment]], Some("A list of apartments")), Unauthorized, BadRequest),
+      response = badRequest.orElse(ok(jsonResponse[List[Apartment]], Some("A list of apartments"))),
```

Comment:
I switched to using `orElse`.

The issue with `withStatusCodes` was that it returned a `Response[ApiError, A]`. However, the type `ApiError` is too wide. For instance, `NotFoundError` is a subtype of `ApiError`, but the endpoint `listApartments` never returns a Not Found response.

This is not a problem in your specific case, but endpoints4s has been designed to generate _clients_ in addition to servers and documentation. From the point of view of a client, calling `listApartments` was previously returning `Either[ApiError, List[Apartment]]`, which means that, in case of `ApiError`, the user would have to deal with the (impossible) case `NotFoundError`.

Another change that I made was to remove the mention of the possible response status `Unauthorized`: this one is automatically handled by the `authenticatedEndpoint` method.

Reply:
Oke, I see your point and it makes perfect sense to change the type to be more specific one.